src/query/interpreter.cpp

```diff
@@ -960,8 +946,8 @@ PullPlan::PullPlan(const std::shared_ptr<CachedPlan> plan, const Parameters &par
   ctx_.evaluation_context.labels = NamesToLabels(plan->ast_storage().labels_, dba);
 #ifdef MG_ENTERPRISE
   if (username.has_value()) {
-    memgraph::auth::User *user = interpreter_context->auth->GetUser(*username);
-    ctx_.fine_grained_access_checker = new FineGrainedAccessChecker{user};
+    ctx_.user = interpreter_context->auth->GetUser(*username);
```

Comment:
User is important because we need to know who is executing query and once we get user we can get `label_permissions ` and `edge_type_permissions` so we can filter what person executing query can or can't access to.

Reply:
Why not return by value?